gridstatus/eia.py

```diff
@@ -54,7 +54,7 @@ def _fetch_page(self, url, headers):
         df = pd.DataFrame(response["data"])
         return df, response["total"]
 
-    def get_dataset(self, dataset, start, end, n_workers=1, verbose=False):
+    def get_dataset(self, dataset, start, end, facets={}, n_workers=1, verbose=False):
```

Comment:
Using mutable default arguments like facets={} is risky in Python. The default value is shared across all calls to the function. If facets is modified in one call, the modified value will be used in subsequent calls, leading to unexpected behavior.

Use None and set to a new dictionary inside the function instead:

```python
def get_dataset(self, dataset, start, end, facets=None, n_workers=1, verbose=False):
    if facets is None:
        facets = {}
```

This way, a new dictionary is created for each function call, avoiding the shared state issue.